### ISSUE TITLE
[NPC] Scarlet Monastery Library NPC bugs.

### ISSUE THREAD

**Issue by highness** [dalaranwow/dalaran-wow#1018]
Here's a list of SM: Library NPCs with notes on what's need to be done with some of them yet.

Scarlet Gallants (http://www.wowhead.com/npc=4287) should have Crusader Strike (http://www.wowhead.com/spell=14517) and Hammer of Justice (http://www.wowhead.com/spell=77787) abilities, currently they use none and will just melee.

Scarlet Beastmasters (http://www.wowhead.com/npc=4288) should use Shoot (http://www.wowhead.com/spell=6660) and Exploding Shot (http://www.wowhead.com/spell=7896) if engaged from distance, and switch to melee if player who has aggro on them closes in. Currently they won't use any abilities and go melee no matter from which distance engaged.

Scarlet Monks (http://www.wowhead.com/npc=4540) - scripted, but I think they use Thrash (http://www.wowhead.com/spell=3391) not as often as they should. Actually, I almost have thought it's missing, but ONE monk near the end of the instance used it at last seemingly at random. Now that I'm thinking on it, many more world mobs should have it, and I almost never saw them using it, so it's probably something either with common AI script or with ability per se.
Also, Kick (http://www.wowhead.com/spell=76583) could be more streamlined, I saw in combat log that Monks were Kicking me even when I didn't cast anything. For the same reason, they can't lock your spells completely - and they should, here's the Wowhead comment: 

"These guys do a kick attack with a cooldown that is shorter than the lockout duration (or equal to it) so dont try to cast in their face or you will get owned hard."

Scarlet Diviners (http://www.wowhead.com/npc=4291) should have 2 abilities: Fireball (http://www.wowhead.com/spell=20815) and Mana Burn (http://www.wowhead.com/spell=11981). Currently they cast Fireball only (though it might be so because I was the only target, and they're scripted to Mana Burn other guys beside the tank).

So, in conclusion, in SM: Library there are problems with Beastmasters and Gallants, and there might be some problems (or there might be not) with Monks and Diviners.

**Comment by Rushor** [comment]
```sql
-- Scarlet Gallant SAI
SET @ENTRY := 4287;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,2,0,100,1,0,50,0,0,11,5589,32,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Gallant - Between 0-50% Health - Cast 'Hammer of Justice' (No Repeat)"),
(@ENTRY,0,1,0,0,0,100,0,2000,5000,12000,26000,11,14517,0,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Gallant - In Combat - Cast 'Crusader Strike'");

-- Scarlet Beastmaster SAI
SET @ENTRY := 4288;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,0,0,100,0,2000,3000,3000,4000,11,6660,0,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Beastmaster - In Combat - Cast 'Shoot'"),
(@ENTRY,0,1,0,4,0,100,0,0,0,0,0,11,7896,0,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Beastmaster - On Aggro - Cast 'Exploding Shot'");

-- Scarlet Monk SAI
SET @ENTRY := 4540;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,13,0,100,2,2000,3000,0,0,11,11978,0,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Monk - On Victim Casting - Cast 'Kick' (Normal Dungeon)"),
(@ENTRY,0,1,0,9,0,100,2,0,2,5000,6000,11,3391,0,0,0,0,0,1,0,0,0,0,0,0,0,"Scarlet Monk - Within 0-2 Range - Cast 'Thrash' (Normal Dungeon)");

-- Scarlet Diviner SAI
SET @ENTRY := 4291;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,0,0,100,2,0,0,2400,3800,11,9053,64,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Diviner - In Combat - Cast 'Fireball' (Normal Dungeon)"),
(@ENTRY,0,1,0,2,0,100,3,0,15,0,0,25,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Scarlet Diviner - Between 0-15% Health - Flee For Assist (No Repeat) (Normal Dungeon)"),
(@ENTRY,0,2,0,0,0,100,0,2000,3000,6000,10000,11,11981,64,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Diviner - In Combat - Cast 'Mana Burn'");
```
DB/Creature: Update some abilities of Scarlet Monastery

* closes https://github.com/dalaranwow/dalaran-wow/issues/1018
* commit: 529f933899f47c344a11dbc5065f3ae672750b43
* all fixed, you can see the spellcombattimer in the script. for example:
```sql
(@ENTRY,0,2,0,0,0,100,0,2000,3000,6000,10000,11,11981,64,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Diviner - In Combat - Cast 'Mana Burn'");
```

means that the creature will start casting Mana Burn after 2-3 seconds after starting combat and repeat it every 6-10 seconds in combat.

The issue is fixed on the PTR Realm. The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Please see and watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by TeridanDW** [comment]
Feedback for these 4 creatures:
 - Scarlet Gallant - works as intended on ptr
Note: I would like to recommend to dramaticly reduce the cooldown of spell http://wotlk.openwow.com/spell=14517 as it does low damage and has high cooldown.

 - Scarlet Beastmaster - Spell http://wotlk.openwow.com/spell=7896 is triggered on aggro and has cast time of 1 sec and almost always fails (because of target been out of range or not in line of sight), not sure if it's intended.

 - Scarlet Monk - Spell http://wotlk.openwow.com/spell=3391 has strange effect ingame (target hits himself):

> Scarlet Monk casts Thrash.
> Scarlet Monk's gains 2 extra attacks through Thrash.
> Scarlet Monk's melee swing hits Scarlet Monk for 108 Physical.
> Scarlet Monk's attack was dodged by Scarlet Monk.
> Scarlet Monk's melee swing hits Teridan for 132 Physical.
> Scarlet Monk's melee swing hits Teridan for 120 Physical.
> Scarlet Monk's melee swing hits Teridan for 154 Physical.
> Scarlet Monk casts Thrash.
> Scarlet Monk's gains 2 extra attacks through Thrash.
> Scarlet Monk's melee swing hits Scarlet Monk for 108 Physical.
> Scarlet Monk's melee swing hits Scarlet Monk for 90 Physical.
> Scarlet Monk's melee swing hits Teridan for 152 Physical.
> Scarlet Monk's melee swing hits Teridan for 138 Physical.
> Scarlet Monk's melee swing hits Teridan for 125 Physical.
> Scarlet Monk casts Thrash.
> Scarlet Monk's gains 2 extra attacks through Thrash.
> Scarlet Monk's melee swing hits Scarlet Monk for 90 Physical.(5 Overkill)
> Scarlet Monk died.

and the range trigger can be avoided if target stays at max melee range (maybe we can use EVENT_UPDATE_IC ?)
To fix the first issue just change:
``` sql
(@ENTRY,0,1,0,9,0,100,2,0,2,5000,6000,11,3391,0,0,0,0,0,1,0,0,0,0,0,0,0,"Scarlet Monk - Within 0-2 Range - Cast 'Thrash' (Normal Dungeon)");
```
to
``` sql
(@ENTRY,0,1,0,9,0,100,2,0,2,5000,6000,11,3391,0,0,0,0,0,2,0,0,0,0,0,0,0,"Scarlet Monk - Within 0-2 Range - Cast 'Thrash' (Normal Dungeon)");
```
 - Scarlet Diviner - works as intended.

**Comment by Rushor** [comment]
I'm going to rework the whole dungeon soon, also adding some texts on aggro

**Comment by Rushor** [comment]
Fixed with https://github.com/dalaranwow/principal/commit/5a9b97d80862d406a3e4bc9358c741f1f792721b

* reduce cooldown of 'Crusader Strike' for Scarlet Gallant to 6000,12000 ms
* Enable cast of 'Exploding Shot' for Scarlet Beastmaster
* Remove mutiple stacking of spell 'Trash' from Scarlet Monk
* Add Aggrotexts for all Scarlet Gallant, Scarlet Beastmaster, Scarlet Monk and Scarlet Diviner
